drivers/SmartThings/matter-switch/src/switch_handlers/attribute_handlers.lua

```diff
@@ -267,45 +270,82 @@ end
 
 -- [[ ELECTRICAL ENERGY MEASUREMENT CLUSTER ATTRIBUTES ]] --
 
-function AttributeHandlers.cumul_energy_imported_handler(driver, device, ib, response)
-  if ib.data.elements.energy then
-    local watt_hour_value = ib.data.elements.energy.value / fields.CONVERSION_CONST_MILLIWATT_TO_WATT
-    device:set_field(fields.TOTAL_IMPORTED_ENERGY, watt_hour_value, {persist = true})
-    if ib.endpoint_id ~= 0 then
-      device:emit_event_for_endpoint(ib.endpoint_id, capabilities.energyMeter.energy({ value = watt_hour_value, unit = "Wh" }))
+function AttributeHandlers.energy_imported_factory(is_periodic_report)
+  return function(driver, device, ib, response)
+    if version.api < 11 then
+      clusters.ElectricalEnergyMeasurement.types.EnergyMeasurementStruct:augment_type(ib.data)
+    end
+    if ib.data.elements.energy then
+      local energy_imported_wh = ib.data.elements.energy.value / 1000 -- convert received milliwatt-hour to watt-hour
+      if is_periodic_report then
+        -- handle this report only if cumulative reports are not supported
+        if device:get_field(fields.CUMULATIVE_REPORTS_SUPPORTED) then return end
+        local state_device = switch_utils.find_child(device, ib.endpoint_id) or device
+        local energy_meter_latest_state = state_device:get_latest_state(
+          "main", capabilities.energyMeter.ID, capabilities.energyMeter.energy.NAME, 0 -- 0 as the default if state is nil
+        )
+        energy_imported_wh = energy_imported_wh + energy_meter_latest_state
+      end
+      device:emit_event_for_endpoint(ib.endpoint_id, capabilities.energyMeter.energy({ value = energy_imported_wh, unit = "Wh" }))
```

Comment:
Should this be `state_device` instead of just `device`? I'm not entirely sure I am following what `state_device` is supposed to be in this case